WordPress/gutenberg
# Global Styles Sidebar: Re-add Colors: Heading to selected blocks

## What?

### WIP

Prior to https://github.com/WordPress/gutenberg/pull/48893,
The panel at `Site Editor Styles > Blocks > Block name > Colors` would show a heading color option for all blocks that supported text color.
This was incorrect, and solved in 48893.

This PR adds the Colors > Heading option back for selected blocks only, through opt-in in block.json.

## Why?


These blocks has or can have headings inside them, and having this option in the Styles sidebar is a useful feature.

## Testing Instructions
1. Activate a theme with support for the site editor.
2. Open the Styles sidebar in the Site Editor.
3. Confirm that root level Styles > Colors > Elements > Headings still work.
4. Select Styles > Blocks > Columns, and confirm that the Colors panel includes Headings.
5. Select Styles > Blocks > Verse, and confirm that the Colors panel do not include Headings.

## Screenshots or screencast 
Before:
The Verse block has a heading color option in the Styles sidebar:
https://user-images.githubusercontent.com/7422055/225591043-2cd43ce7-aaf1-487f-9c3c-0d3f0f8f2413.mov

After:
The Verse block does not have a heading color option in the Styles sidebar:
https://user-images.githubusercontent.com/7422055/233548549-683c8dd2-62ee-4831-a169-7526aebaf05d.mp4
